source/lac/cuda_solver_direct.cu

```diff
@@ -0,0 +1,352 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2018 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#include <deal.II/lac/cuda_solver_direct.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace CUDAWrappers
+{
+  namespace internal
+  {
+    void cusparsecsr2dense(cusparseHandle_t cusparse_handle,
```

Comment:
I guess these are all internal functions you never intend to call from any other .cc file, right? If so, you could wrap them in another level of anonymous namespace so their names don't pollute the linker space.